### ISSUE TITLE
Error when using the UDP protocol

### ISSUE THREAD

**Issue by jsbeltz** [kirm/sip.js#87]
I get an error like:
  TypeError: Cannot set property 'rport' of undefined
      at Object.create.send.value [as send] (F:\builds\users_jeff.beltz_systest\jscript\node_modules\sip\sip.js:773:41)
      at F:\builds\users_jeff.beltz_systest\jscript\node_modules\sip\sip.js:1354:18
      at resolve (F:\builds\users_jeff.beltz_systest\jscript\node_modules\sip\sip.js:838:12)
      at F:\builds\users_jeff.beltz_systest\jscript\node_modules\sip\sip.js:1344:13
      at Object.send (F:\builds\users_jeff.beltz_systest\jscript\node_modules\sip\sip.js:1345:11)

I have added code locally and this resolves the problem.

```javascript
 function wrap(obj, target) {
    return Object.create(obj, {send: {value: function(m) {
      if(m.method) {
        m.headers.via[0].host = options.publicAddress || options.address || options.hostname || os.hostname();
        m.headers.via[0].port = options.port || defaultPort(this.protocol);
        m.headers.via[0].protocol = this.protocol;

        if(this.protocol === 'UDP' && (!options.hasOwnProperty('rport') || options.rport)) {
          // addition
          if(m.headers.via[0].params === undefined)
              m.headers.via[0].params = {};
          // end addition
          m.headers.via[0].params.rport = null;
        }
      }
      options.logger && options.logger.send && options.logger.send(m, target);
      obj.send(m);
    }}});
  }
```

**Comment by kirm** [comment]
And under what circumstance did you get the error? Because your patch just masks the fact that transport received malformed message

**Comment by jsbeltz** [comment]
I think when I send an ACK back like:

    sip.send({
        method: 'ACK',
        uri: rs.headers.contact[0].uri.split(';')[0],
        headers: {
            to: rs.headers.to,
            from: rs.headers.from,
            'call-id': rs.headers['call-id'],
            cseq: {
                method: 'ACK',
                seq: rs.headers.cseq.seq
            },
            contact: [{ uri: contact }],
            via: [{}]
        }

**Comment by jsbeltz** [comment]
So adding to this.  I am generating an INVITE when I get the response, I have to use it to send the ACK, but I should have to build a via. Maybe all it is is user error and I should be doing this differently.

**Comment by kirm** [comment]
You should change your via header from: 

via:[{}]' 

to 

via:[{params: sip.generateBranch()}]

**Comment by jsbeltz** [comment]
Example invite:
    sip.send( { method: 'INVITE'
              , uri: 'session.requestUri
              , headers: { to: session.toUri
                  , from: session.fromUri
                  , 'call-id': session.callid
                  , cseq: { method: session.method, seq: session.cseq }
                  , contact: [{ uri: session.contact }]
                  , 'content-type': 'application/sdp'
                  }
              , content: data.content
              },
              callback)

**Comment by jsbeltz** [comment]
Will try, thanks!

**Comment by jsbeltz** [comment]
That did it thanks.   I hated having that hack in there (my local copy) and I final decided to open a ticket for it.  Thanks again.

**Comment by kirm** [comment]
The real problem here is - some time ago it would be done automatically. Feature seem to have been lost then i added websockets.

**Comment by jsbeltz** [comment]
I haven't tried (yet) using the websockets.   I really like the fact that you have more standard protocols than jssip.   I haven't attempted yet to use TLS but will be soon.

**Comment by kirm** [comment]
I added commit to restore automatic via generation for ACK requests. Now you can drop via header generation from your code completely. Could you please check if it works with your code? Commit 42f71d5d1cb28b5e4d412f38c58fd5f548378937

**Comment by jsbeltz** [comment]
Yes now if I remove the via header for my ACK it works.

**Comment by kirm** [comment]
update published on npm as sip@0.0.2-43
